lab_4_summarization_textrank/main.py

```diff
@@ -287,15 +398,51 @@ def __init__(
         :param punctuation: a sequence of punctuation symbols
         :param idf_values: pre-computed IDF values
         """
-        pass
+        self._stop_words = stop_words
+        self._punctuation = punctuation
+        self._idf_values = idf_values
+        self._text_preprocessor = TextPreprocessor(stop_words, punctuation)
+        self._sentence_encoder = SentenceEncoder()
+        self._sentence_preprocessor = SentencePreprocessor(stop_words, punctuation)
+        self._knowledge_database = {}
+        self._paths_to_texts = paths_to_texts
+
+        for path in paths_to_texts:
+            self.add_text_to_database(path)
 
     def add_text_to_database(self, path_to_text: str) -> None:
         """
         Adds the given text to the existing database
         :param path_to_text
         :return:
         """
-        pass
+        if not path_to_text or not isinstance(path_to_text, str):
+            raise ValueError
+
+        with open(path_to_text, 'r', encoding='utf-8') as file:
```

Comment:
?

Reply:
kirillic alphabet is encoded with utf-8, otherwise we will see something unreadable